### PR TITLE
[random-manager] fail if unable to seed the CTR DRBG

### LINKED ISSUE #6014: [random manager] should we check the return value of mbedtls_ctr_drbg_seed?
Related to #6013, I experienced a hard-to-debug issue, which was masked by the fact that `RandomManager::CryptoCtrDrbg::Init` doesn't check the return value of `mbedtls_ctr_drbg_seed`, leaving the user unaware that the random hardware generator failed.

https://github.com/openthread/openthread/blob/7fe84c59cdf6ce1070ac2d877d826779a04df140/src/core/common/random_manager.cpp#L191-L195

Is this intended? Should we print an error message? Should we assert?

Thanks.

### PR DESCRIPTION
**Pull request by dismirlian**
Resolves #6014.

### CODE REVIEW

File: src/core/common/random_manager.cpp [hunk 1]
```diff
@@ -190,8 +191,19 @@ int RandomManager::Entropy::HandleMbedtlsEntropyPoll(void *         aData,
...
+
     mbedtls_ctr_drbg_init(&mCtrDrbg);
-    mbedtls_ctr_drbg_seed(&mCtrDrbg, mbedtls_entropy_func, RandomManager::GetMbedTlsEntropyContext(), nullptr, 0);
+
+    rval =
+        mbedtls_ctr_drbg_seed(&mCtrDrbg, mbedtls_entropy_func, RandomManager::GetMbedTlsEntropyContext(), nullptr, 0);
+
+    if (rval != 0)
+    {
+        otLogCritMbedTls("Can't seed the CTR DRBG");
```
Comment (jwhui) [inline src/core/common/random_manager.cpp:203]: nit:
```suggestion
        otLogCritMbedTls("Failed to seed the CTR DRBG");
```
I think "fail" is a more common search word when analyzing logs.

**Review by jwhui — COMMENTED** [review]
Thanks! Just one minor nit.

File: src/core/common/random_manager.cpp (hunk 1 above)
Comment (dismirlian) [reply src/core/common/random_manager.cpp]: I agree.

**Review by Irving-cl — APPROVED** [review]
LGTM

**Review by jwhui — APPROVED** [review]

**Pull request merged**